rc/extra/autorestore.kak

```diff
@@ -9,32 +9,36 @@ def autorestore-restore-buffer -docstring "Restore the backup for the current fi
 
         ## Find the name of the latest backup created for the buffer that was open
         ## NB. kak doesn't automatically cd to current buffer_dirname.
-        buffer_backup_glob="${buffer_dirname}/.${buffer_basename}.kak.*"
-        backup_path=$(ls -1t ${buffer_backup_glob} 2>/dev/null | head -n 1)
+        buffer_backup_glob=".${buffer_basename}.kak.*"
+#        backup_path=$(ls -1t ${buffer_backup_glob} 2>/dev/null | head -n 1)
+        newer=$(find "${buffer_dirname}" -maxdepth 1 -name "${buffer_backup_glob}" -newer "${kak_buffile}" 2>/dev/null | head -n1)
+        older=$(find "${buffer_dirname}" -maxdepth 1 -name "${buffer_backup_glob}" \! -newer "${kak_buffile}" 2>/dev/null | head -n1)
 
-        if [ -z "${backup_path}" ]; then
-            exit
-        elif [ ${kak_buffile} -nt ${backup_path} ]; then
-          printf "%s\\n" "echo -debug \"Old backup file found: ${backup_path}; will not restore.\""
-          exit
+
+        if [ -z "${newer}" ];
+        then
+          if [ -n "${older}" ];
+          then
+            printf %s\\n "echo -debug \"Old backup file found: ${older}; will not restore.\""
+          fi
+          exit   
         fi
 
         printf %s\\n "
             ## Replace the content of the buffer with the content of the backup file
-            echo -debug \"Restoring file: ${backup_path}\" 
+            echo -debug \"Restoring file: ${newer}\"
 
-            exec -draft %{ %d!cat<space>${backup_path}<ret>d }
+            exec -draft %{ %d!cat<space>\"${newer}\"<ret>d }
 
             ## If the backup file has to be removed, issue the command once
             ## the current buffer has been saved
             ## If the autorestore_purge_restored option has been unset right after the
             ## buffer was restored, do not remove the backup
             hook -group autorestore buffer BufWritePost '${kak_buffile}' %{
                 nop %sh{
-                    if [ \"\${kak_opt_autorestore_purge_restored}\" = true ]; then
-                        ls -1 '${buffer_dirname}'/.'${buffer_basename}'.kak.* 2>/dev/null | while read -r f; do
-                            rm -f \"\${f}\"
-                        done
+                    if [ \"\${kak_opt_autorestore_purge_restored}\" = true ];
+                    then
+                      rm \"${buffer_dirname}/.${buffer_basename}.kak.\"* 2>/dev/null
```

Comment:
Instead of redirecting `stderr` to `/dev/null`, use the `-f` flag that won't print any error, it's more elegant.